hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequest.java

```diff
@@ -83,7 +83,8 @@ public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
     String keyPath = keyArgs.getKeyName();
 
     OmUtils.verifyKeyNameWithSnapshotReservedWordForDeletion(keyPath);
-    keyPath = validateAndNormalizeKey(ozoneManager.getEnableFileSystemPaths(),
+    keyPath = normalizeKeyPath(ozoneManager.getConfig().isKeyNameCharacterCheckEnabled(), 
+        ozoneManager.getEnableFileSystemPaths(),
```

Comment:
please add a unit test that does:
1. mkdir("/foo:bar")
2. delete("/foo:bar")
This should fail without your patch